### PR TITLE
Nested Attributes

### PR DESCRIPTION
**Pull request by leduytungframgia**

### CODE REVIEW

File: app/views/supervisor/subjects/_subject.html.erb [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
+    <h4 class="panel-title">
+      <a data-toggle="collapse" data-parent="#accordion" href="#<%= subject.id %>"  aria-expanded="true" aria-controls="collapseOne">
+        <%= subject.name %>
+      </a>
+    </h4>
+  </div>
+  <div id="<%= subject.id %>" class="panel-collapse collapse" role="tabpanel" aria-labelledby="headingOne">
+    <div class="panel-body">
+      <%= subject.instruction %>
+      <%= link_to "View" , supervisor_subject_path(subject)%>
```
Comment (NguyenTanDuc) [inline app/views/supervisor/subjects/_subject.html.erb]: %>

File: app/views/supervisor/subjects/show.html.erb [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 <div class="panel panel-info">
-  <div class="alert alert-info" role="alert">
+  <div class="alert alert-info" role="alert"><span class="user">
     <h3><%= link_to @subject.name, @subject %></h3>
     <span class="timestamp">
-      Posted <%= time_ago_in_words @subject.created_at %> ago.
-      <%= link_to "Delete", @subject, method: :delete,
-                                data: {confirm: t(:confirm_notification)} %>
+    Posted <%= time_ago_in_words @subject.created_at %> ago.
+    <%= link_to "Delete", supervisor_subject_path, method: :delete,
```
Comment (NguyenTanDuc) [inline app/views/supervisor/subjects/show.html.erb]: I18n

File: config/routes.rb [hunk 3]
```diff
@@ -11,8 +11,10 @@
 
   namespace :supervisor do
     resources :users
-    resources :subjects
     resources :courses
+    resources :subjects do
+      resources :tasks
+    end
     root  'users#index'
```
Comment (NguyenTanDuc) [inline config/routes.rb]: move to top

File: app/models/subject.rb [hunk 4]
```diff
@@ -1,5 +1,10 @@
 class Subject < ActiveRecord::Base
   has_many :subject_courses
   has_many :courses, through: :subject_courses
-  has_many :tasks
+  has_many :tasks, dependent: :destroy
+  
+  validates :name, presence: true
+  validates :instruction, presence: true
+
+  accepts_nested_attributes_for :tasks, reject_if: lambda {|a| a[:title].blank?}, :allow_destroy => true
```
Comment (NguyenTanDuc) [inline app/models/subject.rb]: replace =>

File: app/helpers/subjects_helper.rb [hunk 5]
```diff
@@ -1,2 +0,0 @@
-module SubjectsHelper
```
Comment (dieunb) [inline app/helpers/subjects_helper.rb]: checkout this file

File: app/views/supervisor/subjects/_form.html.erb [hunk 6]
```diff
@@ -5,8 +5,25 @@
...
-  <%= f.password_field :start_date, class: 'form-control' %>
+  <%= f.text_field :start_date, class: 'form-control' %>
   <%= f.label :end_date, t(:end_date) %>
-  <%= f.password_field :end_date, class: 'form-control' %>
-  <%= f.submit "Submit", class: "btn btn-primary" %>
+  <%= f.text_field :end_date, class: 'form-control' %>
+
+  <%= f.fields_for :tasks do |builder| %>
+    <h2>Task <%= builder.index + 1 %></h2>
+    <%= render "task_fields", :f => builder %>
```
Comment (dieunb) [inline app/views/supervisor/subjects/_form.html.erb]: coding convention

File: app/views/supervisor/subjects/_form.html.erb [hunk 7]
```diff
@@ -5,8 +5,25 @@
...
+  <% end %>
+
+  <%= f.fields_for :tasks, @subject.tasks.build do |builder| %>
+    <div id="task1" class="clonedInput">
+      <h2 id="reference" name="reference" class="heading-reference"> New task </h2>
+      <%= render "task_fields", :f => builder %>
```
Comment (dieunb) [inline app/views/supervisor/subjects/_form.html.erb]: coding conventioncoding convention

File: app/views/supervisor/subjects/_form.html.erb [hunk 8]
```diff
@@ -5,8 +5,25 @@
...
+    </div>
+    <div id="addDelButtons">
+      <input type="button" id="btnAdd" value="Add Task"> 
+      <input type="button" id="btnDel" value="remove section above">
+    </div>
+  <% end %>
+
+    <%= f.submit "Submit", class: "btn btn-primary" %>
```
Comment (dieunb) [inline app/views/supervisor/subjects/_form.html.erb]: indent

**Pull request closed without merging**